Extension modules fail to build

While trying to use a package with extension modules (`py-radix` in particular), I'm seeing the following error ([repro fork](https://github.com/code-clash/rules_pycross/tree/bug/ext-modules)):

```
../bin/clang -shared -Wl,-S -mmacos-version-min=10.11 -no-canonical-prefixes -fobjc-link-runtime -headerpad_max_install_names -lc++ -lm -L../lib build/temp.macosx-11.0-arm64-cpython-311/radix/_radix.o build/temp.macosx-11.0-arm64-cpython-311/radix/_radix/radix.o -o build/lib.macosx-11.0-arm64-cpython-311/radix/_radix.cpython-311-darwin.so
Undefined symbols for architecture arm64:
  "_PyArg_ParseTuple", referenced from:
      _Radix_nodes in _radix.o
      _Radix_prefixes in _radix.o
      _radix_Radix in _radix.o
  "_PyArg_ParseTupleAndKeywords", referenced from:
      _Radix_add in _radix.o
      _Radix_delete in _radix.o
      _Radix_search_exact in _radix.o
      _Radix_search_best in _radix.o
      _Radix_search_worst in _radix.o
      _Radix_search_covered in _radix.o
      _Radix_search_covering in _radix.o
      ...
  "_PyBytes_FromStringAndSize", referenced from:
      _newRadixNodeObject in _radix.o
  "_PyDict_GetItemString", referenced from:
      _module_initialize in _radix.o
<<<OTHER SYMBOLS>>>
ld: symbol(s) not found for architecture arm64
```

It *looks* like the python library is missing from the libs, so I tried using `pycross_wheel_build` with `native_deps = ["@rules_python//python/cc:current_py_cc_libs"]`, but that seems to break even earlier due to wrong includes.

Any ideas on how this can be made to work? Thanks!

--- a/.github/workflows/bazel.yaml
+++ b/.github/workflows/bazel.yaml
@@ -84,8 +84,6 @@ jobs:
         if: (github.ref == 'refs/heads/main' || contains(github.head_ref, 'windows')) && !inputs.exclude_windows
       - id: macos
         run: echo "os=macos-latest" >> $GITHUB_OUTPUT
-        # Only run on main branch (or PR branches that contain 'macos') to minimize macOS minutes (billed at 10X)
-        if: github.ref == 'refs/heads/main' || contains(github.head_ref, 'macos')
     outputs:
       # Will look like ["ubuntu-latest", "windows-latest", "macos-latest"]
       os: ${{ toJSON(steps.*.outputs.os) }}

--- a/pycross/private/tools/wheel_builder.py
+++ b/pycross/private/tools/wheel_builder.py
@@ -340,6 +340,12 @@ def generate_cross_sysconfig_vars(
     sysconfig_vars["LDSHARED"] = " ".join([sysconfig_vars["CC"], sysconfig_vars["LDSHAREDFLAGS"]])
     del sysconfig_vars["LDSHAREDFLAGS"]
 
+    # On macOS, Python C extension shared libraries reference Python API symbols (e.g. PyDict_New)
+    # that are resolved at load time by the interpreter. Tell the linker to allow these undefined
+    # symbols rather than erroring at link time.
+    if sysconfig_vars.get("MACHDEP") == "darwin":
+        sysconfig_vars["LDSHARED"] += " -Wl,-undefined,dynamic_lookup"
+
     # https://github.com/pypa/distutils/issues/283
     sysconfig_vars["LDCXXSHARED"] = sysconfig_vars["LDSHARED"]
 

